### ISSUE TITLE
Allow RC boiler to burn IC2 biogas.

### ISSUE THREAD

**Issue by SpwnX** [CovertJaguar/Railcraft#638]
Well, just as forestry ethanol can be burned in RC boilers, IC2 biogas should be given some love.

Information about IC2 biogas
Energy value of IC2 biogas: 32000 EU per bucket.

**Comment by Forecaster** [comment]
We need the fuel (or heat) value as well

**Comment by SpwnX** [comment]
Heat value for IC2 fluid based heat generator is 64000 hU (heat Units) for 1000 mB
Which then is converted to 32000 EU using a stirling generator.
Or more if you use IC2 (superheated) steam, but thats not the point here.

**Comment by abculatter2** [comment]
Based on a few quick approximate calculations, 3000 Railcraft heat units per bucket should be a good number. Mainly because it's nice and round, and will give an extra 10k eu per bucket. (assuming you use the steam turbine, of course. If you use a 2-to-1 steam-to-eu converter, it would just be an extra 1750 eu)

**Comment by CovertJaguar** [comment]
3000 HU is worse that Creosote at 4800 HU, Ethanol is 16000 HU. BC Fuel is 48000 HU. These are standard vanilla Furnace Heat Units btw.

**Comment by abculatter2** [comment]
Yeah, a bucket of creosote gives 108,000mb steam in a max-heat, size 36 LP boiler. That's 67,500 eu in the turbine, or 54k eu for 2-to-1 steam-to-eu conversion.

Forecaster's boiler calculator says it's actually 106,667mb steam per creosote, but my number is an approximate, so that seems fairly close.

**Comment by CovertJaguar** [comment]
You've lost me.

**Comment by abculatter2** [comment]
I was giving total energy produced for a creosote oil bucket in a max-size, max-heat steam boiler, which from what I understand is the most efficient way to burn any kind of boiler fuel. (Assuming you have an infinite amount of it, of course)
My initial numbers were based on an approximated number for the fuel consumption of such a boiler. (16 fuel units per tick, the actual number is I think 15.8 or something)

I then checked Forecaster's steam boiler calculator, which is more accurate, and it comes out as 106,667mb of steam per bucket of creosote oil.
If you convert this into eu using the 'typical' conversion rate of 2mb steam to 1 eu, this comes out as 53,333.5 eu in total.
In you use the more efficient Railcraft steam turbine, however (1.6 steam to 1 eu) then you get 66,666.875 eu in total.

**Comment by CovertJaguar** [comment]
OK...and how do you get from there to 3000 HU?

**Comment by abculatter2** [comment]
I arrived at 3000 HU for biogas by calculating how much HU would be necessary to make 64,000 mb of steam. I think it ended up being around 2800 HU, and I rounded it up to 3000 HU so that there would be a small bonus from using the boiler.

To be more specific for the process, I did this simple equation;
(64000 / 360) \* 16

**Comment by CovertJaguar** [comment]
Ah ok, I follow now. I've never tried to go from Steam to HU because of how variable the result is. I usually base it on ratios of known quantities. Is there a machine in IC2 that accepts Biogas and something we already have a value for? Ethanol, BC Fuel, etc?

```
private void addLiquidFuels() {
                int bioHeat = (int) (16000 * RailcraftConfig.boilerBiofuelMultiplier());
                Fluid ethanol = Fluids.BIOETHANOL.get();
                if (ethanol != null)
                    FuelManager.addBoilerFuel(ethanol, bioHeat); // Biofuel

                Fluid biofuel = Fluids.BIOFUEL.get();
                if (biofuel != null)
                    FuelManager.addBoilerFuel(biofuel, bioHeat); // Biofuel

                Fluid fuel = Fluids.FUEL.get();
                if (fuel != null)
                    FuelManager.addBoilerFuel(fuel, (int) (48000 * RailcraftConfig.boilerFuelMultiplier())); // Fuel

                Fluid coal = Fluids.COAL.get();
                if (coal != null)
                    FuelManager.addBoilerFuel(coal, (int) (32000 * RailcraftConfig.boilerFuelMultiplier())); // Liquefaction Coal

                Fluid pyrotheum = Fluids.PYROTHEUM.get();
                if (pyrotheum != null)
                    FuelManager.addBoilerFuel(pyrotheum, (int) (64000 * RailcraftConfig.boilerFuelMultiplier())); // Blazing Pyrotheum

                Fluid creosote = Fluids.CREOSOTE.get();
                if (creosote != null)
                    FuelManager.addBoilerFuel(creosote, 4800); // Creosote
            }
```

https://github.com/CovertJaguar/Railcraft/blob/mc-1.10.2/src/main/java/mods/railcraft/common/modules/ModuleCore.java#L384

**Comment by abculatter2** [comment]
The Semifluid Generator; http://wiki.industrial-craft.net/index.php?title=Semifluid_Generator

Also, as SpwnX said, you can get an additional boost by making Superheated Steam instead. Specifically, this gives a 50% boost to power output, assuming I'm reading the numbers correctly. It is a bit more complicated to do though, of course.

**Comment by CovertJaguar** [comment]
If that is list is accurate, that would mean that Biogas and Ethanol should be the same values: 16000 HU.

**Comment by Ristellise** [comment]
yeah cause (Technically) they are the same.

**Comment by CovertJaguar** [comment]
Fixed, assuming that the fluid name is "ic2biogas". I think that was how IC2 registers it, but its hard to tell.
